include/aikido/planner/ConfigurationToEndEffectorOffset.hpp

```diff
@@ -0,0 +1,79 @@
+#ifndef AIKIDO_PLANNER_CONFIGURATIONTOCONFIGURATIONS_HPP_
+#define AIKIDO_PLANNER_CONFIGURATIONTOCONFIGURATIONS_HPP_
+
+#include <dart/dart.hpp>
+#include "aikido/constraint/Testable.hpp"
+#include "aikido/planner/Problem.hpp"
+#include "aikido/statespace/StateSpace.hpp"
+#include "aikido/trajectory/Interpolated.hpp"
+
+namespace aikido {
+namespace planner {
+
+/// Planning problem to plan to desired end-effector offset while maintaining
+/// the current end-effector orientation.
+class ConfigurationToEndEffectorOffset : public Problem
+{
+public:
+  /// Constructor.
+  ///
+  /// \param[in] stateSpace State space.
+  /// \param[in] endEffectorBodyNode BodyNode to be planned to move to a desired
+  /// offest while maintaining the current orientation.
+  /// \param[in] startState Start state.
+  /// \param[in] direction Unit vector that represents the direction of motion
+  /// [unit vector in the world frame]. The direction is allowed to be zero
+  /// vector.
+  /// \param[in] signedDistance Signed distance to move, in meters.
+  /// \param[in] interpolator Interpolator used to produce the output
+  /// trajectory.
+  /// \param[in] constraint Trajectory-wide constraint that must be satisfied.
+  /// \throw (1) If \c stateSpace is not compatible with \c constraint's state
+  /// space.
+  ConfigurationToEndEffectorOffset(
+      statespace::ConstStateSpacePtr stateSpace,
+      dart::dynamics::ConstBodyNodePtr endEffectorBodyNode,
+      const statespace::StateSpace::State* startState,
+      const Eigen::Vector3d& direction,
+      double signedDistance,
```

Comment:
Does this mean that it can have both positive and negative? (We'll have to make sure this is supported in underlying planners.)

Reply:
Yes. If the underlying planners don't support negative distance, they could flip the direction and sign or throw an exception if flipping doesn't make sense to the planner.